packages/edit-post/src/components/header/writing-menu/index.js

```diff
@@ -59,6 +45,7 @@ function WritingMenu() {
 			<PreferenceToggleMenuItem
 				scope="core"
 				name="distractionFree"
+				handleToggling={ false }
```

Comment:
Nit: introducing a new public prop/API for a single use case feels odd. Should we mark it as unstable or experimental?

Reply:
Maybe we should yes, but I think the ability to handle the toggling in the onToggle is not that farfetched.

Reply:
Maybe default toggle shouldn't run when onToggle callback is defined - `onToggle ? onToggle() : toggle()`

cc @talldan

Reply:
I don't think this is right, as the main purpose of this component is a thin wrapper around `MenuItem` that toggles the preference in the preference store, and this code circumvents the one thing it does.

I think an alternative would be to implement it using a `MenuItem`. There's only a spoken message that has to be output, and the distraction free menu item provides its own custom messages anyway, so that logic isn't complicated.

Seems like it'll end up being a very similar amount of code just using a `MenuItem`.